src/solvers/adaptive_time_solver.C

```diff
@@ -87,27 +98,31 @@ void AdaptiveTimeSolver::reinit()
 
 void AdaptiveTimeSolver::advance_timestep ()
 {
-  // The first access of advance_timestep happens via solve, not user code
-  // It is used here to store any initial conditions data
-  if (!first_solve)
-    {
-      // We call advance_timestep in user code after solve, so any solutions
-      // we will be storing will be for the next time instance
-      _system.time += _system.deltat;
-    }
-    else
-    {
-      // We are here because of a call to advance_timestep that happens
-      // via solve, the very first solve. All we are doing here is storing
-      // the initial condition. The actual solution computed via this solve
-      // will be stored when we call advance_timestep in the user's timestep loop
-      first_solve = false;
-    }
+  // // The first access of advance_timestep happens via solve, not user code
+  // // It is used here to store any initial conditions data
+  // //if (!first_solve)
+  //   {
+  //     // We call advance_timestep in user code after solve, so any solutions
+  //     // we will be storing will be for the next time instance
+  //     _system.time += _system.deltat;
+  //   }
+  //   else
+  //   {
+  //     // We are here because of a call to advance_timestep that happens
+  //     // via solve, the very first solve. All we are doing here is storing
+  //     // the initial condition. The actual solution computed via this solve
+  //     // will be stored when we call advance_timestep in the user's timestep loop
+  //     first_solve = false;
+  //   }
```

Comment:
Maybe I read too much into his cleanup commits, but the copy of @jwpeterson who lives in my head is currently frowning at commented-out-but-undeleted code.

If we don't need this anymore (which surprises me, since I thought we'd still have to at least call core_time_solver->advance_timestep() from here; thanks for the comment below) then we should delete it entirely.

Reply:
Regarding the call to core_time_solver->advance_timestep() call, that isnt necessary since the two step time solver does it.
And we need the two step time solver to do the second advance_timestep to make sure we do all the SH store operations before we mess with the deltat.

There is the broader question of the system and user both being able to fiddle with the time, which I think will create problems. Is there a reason the advancing of the timestep cannot always be handled by the timesolver ?

Reply:
Regarding the commenting, wanted to run this by you before I deleted those lines. We should be clear to remove them now.

Reply:
Yeah, I think ideally advancing the time would only be done by advance_timestep().  Isn't that how it works now?  I don't see system.time being touched outside of advance_timestep().

Reply:
The copy of me living in @roystgnr's head:

![ralph](https://user-images.githubusercontent.com/1775907/85593708-e40ad780-b60c-11ea-9ae8-c6cdacf61732.jpg)

Reply:
Sorry, GitHub hasn't been updating pages on me properly.  Definitely clear to remove.